### PR TITLE
feat(driver): move some components into separated directories

### PR DESCRIPTION
**Pull request by davidtaikocha**

### CODE REVIEW

File: driver/anchor_tx_constructor/anchor_tx_constructor_test.go [hunk 1]
```diff
@@ -41,7 +62,7 @@ func (s *DriverTestSuite) TestSign() {
 	s.Nil(err)
 	isVsalid := crypto.VerifySignature(pubKey, hash, signatureBytes[:64])
```
Comment (alexshliu) [inline driver/anchor_tx_constructor/anchor_tx_constructor_test.go:63]:
```suggestion
	isValid := crypto.VerifySignature(pubKey, hash, signatureBytes[:64])
```

Comment (davidtaikocha) [reply driver/anchor_tx_constructor/anchor_tx_constructor_test.go]: changed

File: driver/state/state_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+func (s *DriverStateTestSuite) SetupTest() {
+	s.ClientTestSuite.SetupTest()
+	state, err := New(context.Background(), s.RpcClient)
+	s.Nil(err)
+	s.s = state
+}
+
+func (s *DriverStateTestSuite) TestVerfiyL2Block() {
```
[the remark this answers is not in the document]
Comment (davidtaikocha) [reply driver/state/state_test.go]: changed

**Review by alexshliu — APPROVED** [review]

File: driver/chain_syncer/chain_syncer.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+func (s *L2ChainSyncer) AheadOfProtocolVerifiedHead() bool {
+	verifiedHeightToCompare := s.state.GetLatestVerifiedBlock().Height.Uint64()
+	log.Debug(
+		"Checking whether the execution engine is ahead of protocol's verified head",
+		"latestVerifiedBlock", verifiedHeightToCompare,
+		"executionEngineHead", s.state.GetL2Head().Number,
+	)
+	if verifiedHeightToCompare > 0 {
+		// If latest verified head height is equal to L2 execution engine's synced head height minus one,
+		// we also mark the triggered P2P sync progress as finished to prevent a protenial `InsertBlockWithoutSetHead` in
```
[the remark this answers is not in the document]
Comment (davidtaikocha) [reply driver/chain_syncer/chain_syncer.go]: changed

**Pull request merged**